azure-iot-common/tests/test_async_adapter.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+import inspect
+import azure.iot.common.async_adapter as async_adapter
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture
+def dummy_value():
+    return 123
+
+
+@pytest.fixture
+def mock_function(mocker, dummy_value):
+    mock_fn = mocker.MagicMock(return_value=dummy_value)
+    mock_fn.__doc__ = "docstring"
+    return mock_fn
+
+
+async def test_emulate_async(mocker, mock_function, dummy_value):
+    async_fn = async_adapter.emulate_async(mock_function)
+    result = await async_fn(dummy_value)
+    assert inspect.iscoroutinefunction(async_fn)
+    assert async_fn.__doc__ == mock_function.__doc__  # verify docstring carries over
```

Comment:
Too specific an implementation detail IMO. There's other ways to make it happen other than functools.wrap, and I'd rather just test that it has the effect we want, rather than the method by which we achieve it.

---
In reply to: [251156400](https://github.com/Azure/azure-iot-sdk-python-preview/pull/43#discussion_r251156400) [](ancestors = 251156400)